x/ccv/child/keeper/distribution.go

```diff
@@ -0,0 +1,69 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	clienttypes "github.com/cosmos/ibc-go/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/modules/core/04-channel/types"
+	"github.com/cosmos/interchain-security/x/ccv/child/types"
+	// XXX delete or uncomment this block before merge
+	//ibctypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+)
+
+// Simple model, donate tokens to the fee pool of the provider validator set
+// reference: cosmos/ibc-go/modules/apps/transfer/keeper/msg_server.go
+func (k Keeper) DonateToProviderValidatorSet(ctx sdk.Context) error {
+	if !k.shouldTransmit(ctx) {
+		return nil
+	}
+
+	port := k.GetDistributionTransmissionPort(ctx)
+	channel := k.GetDistributionTransmissionChannel(ctx)
+
+	// work around to reuse the IBC token transfer logic
+	consumerFeePoolAddr := k.accountKeeper.GetModuleAccount(ctx, k.feeCollectorName).GetAddress()
+	providerFeePoolAddrStr := k.GetProviderFeePoolAddrStr(ctx)
```

Comment:
We need to document very clearly that `providerFeePoolAddress` is not included as a blocked address in bank keeper. Otherwise, any transfer to it will be blocked

Reply:
Okay yeah good point, haven't dealt with much of this blocked address business. Blocked addresses are configured in the `app.go` if I'm not mistaken. Leaving this comment open for now

Reply:
Added comments in app.go and provider module.go